src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -742,23 +742,35 @@ private BoundExpression BindDeclarationVariables(TypeSymbol declType, VariableDe
                 case SyntaxKind.ParenthesizedVariableDesignation:
                     {
                         var tuple = (ParenthesizedVariableDesignationSyntax)node;
-                        var builder = ArrayBuilder<BoundExpression>.GetInstance(tuple.Variables.Count);
+                        int count = tuple.Variables.Count;
+                        var builder = ArrayBuilder<BoundExpression>.GetInstance(count);
+                        var namesBuilder = ArrayBuilder<string>.GetInstance(count);
+
                         foreach (var n in tuple.Variables)
                         {
                             builder.Add(BindDeclarationVariables(declType, n, n, diagnostics));
+                            namesBuilder.Add(InferTupleElementName(n));
                         }
                         var subExpressions = builder.ToImmutableAndFree();
 
+                        var uniqueFieldNames = PooledHashSet<string>.GetInstance();
+                        RemoveDuplicateInferredTupleNames(namesBuilder, uniqueFieldNames);
+                        uniqueFieldNames.Free();
+
+                        var tupleNames = namesBuilder.ToImmutableAndFree();
+                        var inferredPositions = tupleNames.SelectAsArray(n => n != null);
+
                         // We will not check constraints at this point as this code path
                         // is failure-only and the caller is expected to produce a diagnostic.
                         var tupleType = TupleTypeSymbol.Create(
                             null,
                             subExpressions.SelectAsArray(e => e.Type),
                             default(ImmutableArray<Location>),
-                            default(ImmutableArray<string>),
+                            tupleNames,
                             Compilation,
-                            shouldCheckConstraints: false);
-                        return new BoundTupleLiteral(syntax, default(ImmutableArray<string>), subExpressions, tupleType);
+                            shouldCheckConstraints: false,
+                            inferredPositions: inferredPositions);
+                        return new BoundTupleLiteral(syntax, default(ImmutableArray<string>), default(ImmutableArray<bool>), subExpressions, tupleType);
```

Comment:
>return [](start = 24, length = 6)

Perhaps inferredPositions should be handled here and in the type as per the version elsewhere.